Raven.Database/Indexing/IndexingExecuter.cs

```diff
@@ -258,17 +271,24 @@ private int ExecuteTask(HashSet<int> indexIds, HashSet<IComparable> alreadySeen)
             return processedKeys;
         }
 
-        private DatabaseTask GetApplicableTask(IStorageActionsAccessor actions, HashSet<IComparable> alreadySeen)
+        private DatabaseTask GetApplicableTask(IStorageActionsAccessor actions, ExecutedTasksInfo executedTasksInfo)
         {
             var disabledIndexIds = context.IndexStorage.GetDisabledIndexIds();
 
-            var removeFromIndexTasks = actions.Tasks.GetMergedTask<RemoveFromIndexTask>(
-                disabledIndexIds, context.IndexStorage.Indexes, alreadySeen);
-            if (removeFromIndexTasks != null)
-                return removeFromIndexTasks;
+            if (executedTasksInfo.SkipRemoveFromIndexTask == false)
```

Comment:
Why do we need this optimization? Can it be time consuming?

Reply:
Yes. When we have disabled indexes, we are not going to execute the RemoveFromIndexTask.
If we have only TouchReferenceDocumentIfChangedTask we need to search for RemoveFromIndexTask although we didn't find any at the beginning.
Skipping tasks might be expensive if we have thousands of those.

Reply:
Okay. Thanks for explanation.